src/libstd/sys/unix/thread.rs

```diff
@@ -206,6 +207,29 @@ pub unsafe fn create(stack: uint, p: Thunk) -> rust_thread {
     native
 }
 
+#[cfg(any(target_os = "linux", target_os = "android"))]
+pub unsafe fn set_name(name: &str) {
+    // Using prctl() rather than pthread_setname_np(),
+    // because pthread_setname_np() wasn't added until glibc 2.12
+    // PR_SET_NAME since Linux 2.6.9
+    let cname = CString::from_slice(name.as_bytes());
+    prctl(15i32 /* = PR_SET_NAME */, cname.as_ptr() as u64, 0u64, 0u64, 0u64);
```

Comment:
For `pthread_setname_np`, but largely only because it looks like it's only defined on newer glibc implementations. This is a bit of a nicety, however, so we probably won't lose too much if it's not available.